pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -491,49 +526,64 @@ func waitForQuorumWithMons(context *clusterd.Context, clusterName string, mons [
 			continue
 		}
 
+		if !requireAllInQuorum {
+			logQuorumMembers(monStatusResp)
+			break
+		}
+
 		// check if each of the initial monitors is in quorum
 		allInQuorum := true
 		for _, name := range mons {
-			// first get the initial monitors corresponding mon map entry
-			var monMapEntry *client.MonMapEntry
-			for i := range monStatusResp.MonMap.Mons {
-				if name == monStatusResp.MonMap.Mons[i].Name {
-					monMapEntry = &monStatusResp.MonMap.Mons[i]
-					break
-				}
-			}
-
-			if monMapEntry == nil {
-				// found an initial monitor that is not in the mon map, bail out of this retry
-				logger.Warningf("failed to find initial monitor %s in mon map", name)
-				allInQuorum = false
-				break
-			}
-
-			// using the current initial monitor's mon map entry, check to see if it's in the quorum list
-			// (a list of monitor rank values)
-			inQuorumList := false
-			for _, q := range monStatusResp.Quorum {
-				if monMapEntry.Rank == q {
-					inQuorumList = true
-					break
-				}
-			}
-
-			if !inQuorumList {
+			if !monFoundInQuorum(name, monStatusResp) {
 				// found an initial monitor that is not in quorum, bail out of this retry
-				logger.Warningf("initial monitor %s is not in quorum list", name)
+				logger.Warningf("monitor %s is not in quorum list", name)
 				allInQuorum = false
 				break
 			}
 		}
 
 		if allInQuorum {
-			logger.Debugf("all initial monitors are in quorum")
+			logQuorumMembers(monStatusResp)
 			break
 		}
 	}
 
-	logger.Infof("Ceph monitors formed quorum")
 	return nil
 }
+
+func logQuorumMembers(monStatusResp client.MonStatusResponse) {
+	var monsInQuorum []string
+	for _, m := range monStatusResp.MonMap.Mons {
+		if monFoundInQuorum(m.Name, monStatusResp) {
+			monsInQuorum = append(monsInQuorum, m.Name)
+		}
+	}
+	logger.Infof("Monitors are in quorum: %v", monsInQuorum)
```

Comment:
Nit: I think "Monitors in quorum" makes more sense, as "are in quorum" might suggest that all are in quorum to some users.